src/main/java/com/spring/test/service/UserServiceImpl.java

```diff
@@ -0,0 +1,23 @@
+package com.spring.test.service;
+
+import com.spring.test.dao.UserDao;
+import com.spring.test.model.User;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class UserServiceImpl implements UserService {
+    @Autowired
+    private UserDao userDao;
```

Comment:
What is better: setter auto wiring, constructor auto wiring, or field injection?

Reply:
constructor